apps/wallet/tests/sites-to-cs-messaging.spec.ts

```diff
@@ -0,0 +1,180 @@
+                        msg.data.payload.id === anId
+                    ) {
+                        window.removeEventListener('message', callBackFN);
+                        if (msg.data.payload.payload.error) {
+                            reject(msg.data.payload);
+                        } else {
+                            resolve(msg.data.payload);
+                        }
+                    }
+                };
+                window.addEventListener('message', callBackFN);
+            }),
+        id
+    );
+}
+
+test.beforeEach(async ({ page, extensionUrl }) => {
+    await createWallet(page, extensionUrl);
+    await page.close();
+});
+
+test.describe('site to content script messages', () => {
+    const allTests = [
+        ['get accounts', { type: 'get-account' }, false],
+        [
+            'execute transaction no account',
+            {
+                type: 'execute-transaction-request',
+            },
+            false,
+        ],
+        [
+            'sign transaction no account',
+            {
+                type: 'sign-transaction-request',
+            },
+            false,
+        ],
+        [
+            'sign message no account',
+            {
+                type: 'sign-message-request',
+            },
+            false,
+        ],
+        [
+            'UI get-features',
+            {
+                type: 'get-features',
+            },
+            false,
+        ],
+        [
+            'UI create wallet',
+            {
+                type: 'keyring',
+                method: 'create',
+                args: {},
+            },
+            false,
+        ],
+    ] as const;
+    for (const [aLabel, aPayload, result] of allTests) {
+        test(aLabel, async ({ context, demoPageUrl }) => {
+            const page = await context.newPage();
+            await page.goto(demoPageUrl);
+            const nextMessage = getInAppMessage(page, aLabel);
+            await page.evaluate(
+                ({ aPayload: payload, aLabel: label }) => {
+                    window.postMessage({
+                        target: 'sui_content-script',
+                        payload: {
+                            id: label,
+                            payload,
+                        },
+                    });
+                },
+                { aPayload, aLabel }
+            );
+            if (result) {
+                expect(await nextMessage).toMatchObject(result);
+            } else {
+                await expect(nextMessage).rejects.toThrow();
+            }
+        });
+    }
+});
+
+test.describe('Wallet interface', () => {
+    let demoPage: Page;
+
+    test.beforeEach(async ({ context, demoPageUrl }) => {
+        demoPage = await context.newPage();
+        await demoPage.goto(demoPageUrl);
+    });
+    test.describe('when not connected', () => {
+        // eslint-disable-next-line no-empty-pattern
+        test('no account is connected', async ({}) => {
```

Comment:
nit: can we do `async () => { ...` and remove the linter comment, or is there a reason why we have to do it like this?

Reply:
Removed it seems not necessary here